src/supervisedoutputconnector.h

```diff
@@ -242,13 +245,11 @@ namespace dd
      * @param ad_out output data object
      * @param bcats supervised output connector
      */
-    void best_cats(const APIData &ad_out, SupervisedOutput &bcats,
+    void best_cats(SupervisedOutput &bcats, const int &output_param_best,
                    const int &nclasses, const bool &has_bbox,
                    const bool &has_roi, const bool &has_mask) const
     {
-      int best = _best;
-      if (ad_out.has("best"))
-        best = ad_out.get("best").get<int>();
+      int best = output_param_best;
```

Comment:
Here the behavior changes. Before DTOs the parameter could be set during `init`, and can be modified in predict call if specified. Now that we added a default value, we cannot know if the user specified this parameter or not during predict, so we modify it every time.

Is it fine like this? Or should we stick to the old behavior? I'm not sure how to do with DTOs but it must be possible to know if the user specified a value or not (probably by removing the default value at predict time)

Reply:
A solution might be to have a `default_best` and `user_best` and to use `default_best` when `user_best` hasn't been set ? or any similar mechanism.
Otherwise there's the risk that the last set value does stick, whereas `predict` calls should not permanently modify the resource that is the service.